plugins/aws/resoto_plugin_aws/resource/elasticbeanstalk.py

```diff
@@ -0,0 +1,125 @@
+    }
+    max_count_rule: Optional[AwsBeanstalkMaxCountRule] = field(default=None)
+    max_age_rule: Optional[AwsBeanstalkMaxAgeRule] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsBeanstalkApplicationResourceLifecycleConfig:
+    kind: ClassVar[str] = "aws_beanstalk_application_resource_lifecycle_config"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "service_role": S("ServiceRole"),
+        "version_lifecycle_config": S("VersionLifecycleConfig")
+        >> Bend(AwsBeanstalkApplicationVersionLifecycleConfig.mapping),
+    }
+    service_role: Optional[str] = field(default=None)
+    version_lifecycle_config: Optional[AwsBeanstalkApplicationVersionLifecycleConfig] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsBeanstalkApplication(AwsResource):
+    kind: ClassVar[str] = "aws_beanstalk_application"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("elasticbeanstalk", "describe-applications", "Applications")
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("ApplicationName"),
+        "name": S("ApplicationName"),
+        "ctime": S("DateCreated"),
+        "mtime": S("DateUpdated"),
+        "arn": S("ApplicationArn"),
+        "description": S("Description"),
+        "versions": S("Versions", default=[]),
+        "configuration_templates": S("ConfigurationTemplates", default=[]),
+        "resource_lifecycle_config": S("ResourceLifecycleConfig")
+        >> Bend(AwsBeanstalkApplicationResourceLifecycleConfig.mapping),
+    }
+    arn: Optional[str] = field(default=None)
+    description: Optional[str] = field(default=None)
+    versions: List[str] = field(factory=list)
+    configuration_templates: List[str] = field(factory=list)
+    resource_lifecycle_config: Optional[AwsBeanstalkApplicationResourceLifecycleConfig] = field(default=None)
+
+    def _set_tags(self, client: AwsClient, tags: Dict[str, str]) -> bool:
+        tag_set = [{"Key": k, "Value": v} for k, v in tags.items()]
+        client.call(
+            service="elasticbeanstalk",
+            action="update-tags-for-resource",
+            result_name=None,
+            ResourceArn=self.arn,
+            Tagging={"TagSet": tag_set},
+        )
+        return True
+
+    def _get_tags(self, client: AwsClient) -> Dict[str, str]:
+        """Fetch the Elasticbeanstalk Application tags from the AWS API."""
+        tags: Dict[str, str] = {}
+        try:
+            response = client.call(
+                service="elasticbeanstalk", action="list-tags-for-resource", result_name="TagSet", ResourceArn=self.arn
+            )
+            tags = tags_as_dict(response)  # type: ignore
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] != "NoSuchTagSet":
+                raise
+        return tags
+
+    def update_resource_tag(self, client: AwsClient, key: str, value: str) -> bool:
+        tags = self._get_tags(client)
```

Comment:
I think we can make it simpler: not query the existing tags but call the "update_tags_for_resource" action directly.